core/src/main/java/org/apache/accumulo/core/client/admin/compaction/CompactionConfigurer.java

```diff
@@ -50,6 +51,12 @@ public interface InitParameters {
   public interface InputParameters {
     TableId getTableId();
 
+    /**
+     * @return the id of the tablet being compacted
+     * @since 2.1.1
+     */
+    TabletId getTabletId();
+
```

Comment:
This is an API addition. It shouldn't be added in a patch release, according to our semver rules.

Reply:
Doh.  Right. I was thinking this was SPI, was not paying close attention to the packages.

Reply:
We can hold off on merging this for now or push it to 3.0.0.  I'm wondering how many additive API/SPI/prop refinements will materialize as 2.1.0 is used.  If enough materialize, then it may be a good reason to release 2.2.0.

Reply:
> If enough materialize, then it may be a good reason to release 2.2.0.

If we didn't have the LTM upgrade paths, and were operating under pre-LTM, then I'd agree. However, LTM expectations makes that more complicated. Would users move from 2.1 LTM to 2.2 non-LTM for this? If users are willing to go to non-LTM, why wouldn't they just accept getting these features in 3.x? Are we expecting to maintain another 2.x LTM release so users can get these, plus stability patches? If we're still working on 3.x, how many branches are we now on the hook to maintain? So far, I don't see much compelling here to warrant more branches to maintain, or for users to be motivated to move off an LTM release. However, they might decide to backport something like this to their own internal 2.1 LTM-based fork.

For now, I'd suggest basing this off of the main branch. If we do decide to have another 2.x version, we can backport it from 3.x to a newly created branch for that purpose (but I hope we don't because it's too much overhead to maintain so many different branches and clearly communicate expectations to users about what they get with each branch).

Reply:
Yeah this alone does not warrant a 2.2.0.  Lots of minor changes like this could motivate a 2.2.0 and with proper communication that could be communicated as a continuation of the LTM release that started with 2.1.0, if we wanted to.  However that's speculation based on what may come. Can wait and see what else comes up.

Reply:
How should we track things that could go into a possible 2.2.0 ?

Reply:
I don't think we should do it at all. I don't think we're ever going to have enough to make it compelling. So, if you're motivated to track that, I would suggest you track it for yourself, and then if you feel it has reached a critical point to become compelling, you could list the changes you'd like backported into a 2.2, as part of a mailing list discussion you initiate.

If you really want to track it using the project's issue tracker, then the best way to do that, is to create a project, just like how we have projects for different releases, and just call it "Backport candidate for potential 2.2" or something.

Reply:
> How should we track things that could go into a possible 2.2.0 ?


We could start with an issue that list things w/ links.  If enough things show up there then we could create a project.  Just thinking an issue is much less noisy than a project, especially if it may not happen.  I was also thinking of a gist, but that moves off of community owned infra.

Reply:
I created #3092